src/combat.cpp

```diff
@@ -731,16 +732,16 @@ void Combat::doCombat(Creature* caster, const Position& position) const
 
 		postCombatEffects(caster, position, params);
 
-		for (Tile* tile : tiles) {
+		for (const auto& tile : tiles) {
 			if (canDoCombat(caster, tile, params.aggressive) != RETURNVALUE_NOERROR) {
 				continue;
 			}
 
 			combatTileEffects(spectators, caster, tile, params);
 
 			if (CreatureVector* creatures = tile->getCreatures()) {
-				const Creature* topCreature = tile->getTopCreature();
-				for (Creature* creature : *creatures) {
+				auto topCreature = tile->getTopCreature();
```

Comment:
Could this be `const auto&`
```suggestion
				const auto& topCreature = tile->getTopCreature();
```